server/http/controller.go

```diff
@@ -514,3 +530,89 @@ func (c *controller) executeTest(ctx context.Context, test model.Test, runInfo o
 	}
 	return openapi.Response(200, res), nil
 }
+
+// Environments
+
+func (c *controller) CreateEnvironment(ctx context.Context, in openapi.Environment) (openapi.ImplResponse, error) {
+	environment := c.mappers.In.Environment(in)
+
+	if environment.ID != "" {
+		exists, err := c.testDB.EnvironmentIDExists(ctx, environment.ID)
+
+		if err != nil {
+			return handleDBError(err), err
+		}
+
+		if exists {
+			err := fmt.Errorf(`cannot create environment with ID "%s: %w`, environment.ID, errTestExists)
+			r := map[string]string{
+				"error": err.Error(),
+			}
+			return openapi.Response(http.StatusBadRequest, r), err
+		}
+	}
+
+	environment, err := c.testDB.CreateEnvironment(ctx, environment)
+	if err != nil {
+		return openapi.Response(http.StatusInternalServerError, err.Error()), err
+	}
+
+	return openapi.Response(200, c.mappers.Out.Environment(environment)), nil
+}
+
+func (c *controller) DeleteEnvironment(ctx context.Context, environmentId string) (openapi.ImplResponse, error) {
+	environment, err := c.testDB.GetEnvironment(ctx, environmentId)
+	if err != nil {
+		return handleDBError(err), err
+	}
+
+	err = c.testDB.DeleteEnvironment(ctx, environment)
+	if err != nil {
+		return openapi.Response(http.StatusInternalServerError, err.Error()), err
+	}
```

Comment:
Maybe you should use `handleDBError` instead.

Reply:
Done 😄